home_audit.node.js

```diff
@@ -1227,6 +1221,90 @@ let validationRules = {
         }
     },
     
+    /**
+     * Get validations for wall values, ensuring wall is valid
+     * @param value
+     * @param {string} side
+     * @param {Validation} validation
+     */
+    _get_wall_validation: function(value, side, validation) {
+        const invalidWall = this._is_valid_wall_side(value, side);
+        if ((validation && validation['type'] !== BLOCKER)) {
+            return validation;
+        }
+        if (invalidWall && invalidWall['message']) {
+            return invalidWall;
+        }
+        return validation;
+    },
+    
+    /**
+     * Checks that entered value is on a valid hvac
+     * @param value
+     * @param {int} num
+     */
+    _is_servicing_system: function(value, num) {
+        if(_homeValues['hvac_fraction_'+num] === 0 && !TypeRules._is_empty(value)) {
+            return new Validation(
+                'Values may not be defined for system that do not service any area of the home. Please only set values for hvacs with hvac fraction > 0',
+                ERROR
+            );
+        }
+    },
+    
+    /**
+     * Checks that entered value is on a valid hvac
+     * @param value
+     * @param {int} sys
+     * @param {int} duct
+     */
+    _is_servicing_duct: function(value, sys, duct) {
+        if(_homeValues['duct_fraction_'+duct+'_'+sys] === 0 && !TypeRules._is_empty(value)) {
+            return new Validation(
+                'Values may not be defined for ducts that do not service any area of the home. Please only set values for ducts with duct fraction > 0',
+                ERROR
+            );
+        }
+    },
+    
+    /**
+     * Get validations for wall values, ensuring wall is valid
+     * @param value
+     * @param {string} side
+     * @param {Validation} validation
+     */
+    _get_system_validation: function(value, num, validation) {
+        const invalidSystem = this._is_servicing_system(value, num);
+        if ((validation && validation['type'] !== BLOCKER)) {
+            return validation;
+        }
+        if (invalidSystem && invalidSystem['message']) {
+            return invalidSystem;
+        }
+        return validation;
+    },
+    
+    /**
+     * Get validations for wall values, ensuring wall is valid
+     * @param value
+     * @param {string} side
+     * @param {Validation} validation
+     */
+    _get_duct_validation: function(value, sys, duct, validation) {
+        const invalidSystem = this._is_servicing_system(value, sys);
```

Comment:
It's not a  huge deal but it would be a bit more sensible in all of these not to retrieve the result from _is_servicing_system() before checking if you're just going to return `validation` anyway.